Core/Core/Inbox/AttachmentPicker/Model/AttachmentPickerInteractor/AttachmentPickerInteractorLive.swift

```diff
@@ -68,10 +106,87 @@ class AttachmentPickerInteractorLive: AttachmentPickerInteractor {
                 uploadManager.cancel(file: file)
             }
         }
+        var filteredFiles: [File] = alreadySelectedFiles.value.filter { $0.isUploading == false}
+        alreadySelectedFiles.send(filteredFiles)
     }
 
     func removeFile(file: File) {
+        if alreadySelectedFiles.value.contains(file) {
+            let newValues = alreadySelectedFiles.value.filter { $0 != file }
+            alreadySelectedFiles.send(newValues)
+        } else {
+            uploadManager.viewContext.delete(file)
+            fileStore.refresh()
+        }
+    }
+
+    func deleteFile(file: File) -> AnyPublisher<Void, Never> {
         uploadManager.viewContext.delete(file)
         fileStore.refresh()
+
+        if let fileId = file.id {
```

Comment:
Could you please replace with `guard`?
- Same for the next method below
- and for the map inside that one